src/app/address/controllers/address.js

```diff
@@ -84,7 +99,34 @@ class AddressController extends BaseController {
       validFrom: addressYearFrom,
     };
 
-    return address;
+    const isChanged = this.checkForChanges(address, chosenAddress);
+    if (isChanged) {
+      delete chosenAddress.uprn;
```

Comment:
This mutates chosenAddress, so I wonder if something like:
```
updatedAddress = {
uprn: isChanged ? chosenAddress.uprn : undefined
  ...chosenAddress
}
```
is an improvement?